<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BinSrchViewClass</class>
 <widget class="QWidget" name="BinSrchViewClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1098</width>
    <height>530</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QTextBrowser" name="binSrchInfoTextBrowser">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>330</y>
     <width>371</width>
     <height>191</height>
    </rect>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:10px; margin-bottom:11px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; line-height:160%;&quot;&gt;&lt;span style=&quot; font-family:'Nunito','sans-serif'; font-size:10pt; font-weight:600; font-style:italic; color:#000000;&quot;&gt;Binary search&lt;/span&gt;&lt;span style=&quot; font-family:'Nunito','sans-serif'; font-size:10pt; font-style:italic; color:#000000;&quot;&gt; is a fast search algorithm with run-time complexity of Ο(log n)&lt;/span&gt;&lt;span style=&quot; font-family:'Nunito','sans-serif'; font-size:8pt; color:#000000;&quot;&gt;. This search algorithm works on the principle of divide and conquer. For this algorithm to work properly, the data collection should be in the sorted form.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:10px; margin-bottom:11px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; line-height:160%;&quot;&gt;&lt;span style=&quot; font-family:'Nunito','sans-serif'; font-size:8pt; color:#000000;&quot;&gt;Binary search looks for a particular item by comparing the middle most item of the collection. If a match occurs, then the index of item is returned. If the middle item is greater than the item, then the item is searched in the sub-array to the left of the middle item. Otherwise, the item is searched for in the sub-array to the right of the middle item. This process continues on the sub-array as well until the size of the subarray reduces to zero.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:10px; margin-bottom:11px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; line-height:160%;&quot;&gt;&lt;span style=&quot; font-family:'Heebo','sans-serif'; font-size:8pt; font-weight:496; color:#000000;&quot;&gt;Pseudocode&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:10px; margin-bottom:11px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; line-height:160%;&quot;&gt;&lt;span style=&quot; font-family:'Nunito','sans-serif'; font-size:8pt; color:#000000;&quot;&gt;The pseudocode of binary search algorithms should look like this −&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#660066;&quot;&gt;Procedure&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; binary_search&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   A &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;←&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; sorted array&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   n &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;←&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; size &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;of&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; array&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   x &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;←&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;value&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; to be searched&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Courier New'; font-size:8pt; color:#000000; background-color:#eeeeee;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#660066;&quot;&gt;Set&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; lowerBound &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#006666;&quot;&gt;1&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#660066;&quot;&gt;Set&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; upperBound &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; n &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Courier New'; font-size:8pt; color:#000000; background-color:#eeeeee;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;while&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; x &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;not&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; found&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;      &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;if&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; upperBound &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;&amp;lt;&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; lowerBound &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;         EXIT&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;:&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; x does &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;not&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; exists&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;      &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;set&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; midPoint &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; lowerBound &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;+&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;(&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; upperBound &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;-&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; lowerBound &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;)&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;/&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#006666;&quot;&gt;2&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;      &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;      &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;if&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; A&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;[&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;midPoint&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;]&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;&amp;lt;&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; x&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;         &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;set&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; lowerBound &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; midPoint &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;+&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#006666;&quot;&gt;1&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;         &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;      &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;if&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; A&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;[&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;midPoint&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;]&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;&amp;gt;&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; x&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;         &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;set&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; upperBound &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; midPoint &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;-&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#006666;&quot;&gt;1&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Courier New'; font-size:8pt; color:#000000; background-color:#eeeeee;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;      &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;if&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; A&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;[&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;midPoint&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;]&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; x &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;         EXIT&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;:&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; x found at location midPoint&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;end&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;while&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;end&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; procedure&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="Line" name="binSrchAnimTextAreaSeparator">
   <property name="geometry">
    <rect>
     <x>695</x>
     <y>110</y>
     <width>31</width>
     <height>411</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>691</width>
     <height>97</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="binSrchHeaderAndControlPane">
    <item>
     <widget class="QLabel" name="binSrchHeaderLabel">
      <property name="layoutDirection">
       <enum>Qt::RightToLeft</enum>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:11pt; font-weight:600;&quot;&gt;BINARY SEARCH&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="binSrchControlPane">
      <item>
       <widget class="QLineEdit" name="binSrchInsertLineEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="binSrchSrchBtn">
        <property name="text">
         <string>Search</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="binSrchSrchNewValSepLine">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="binSrchNewValBtn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>New Values</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QPlainTextEdit" name="binSrchPlainTextEdit">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>130</y>
     <width>371</width>
     <height>171</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="binSrchConsoleLabel">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>110</y>
     <width>47</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Console:</string>
   </property>
  </widget>
  <widget class="QWidget" name="binSrchAnimAreaPane" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>110</y>
     <width>691</width>
     <height>411</height>
    </rect>
   </property>
   <widget class="QGraphicsView" name="binSrchAnimArea">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>701</width>
      <height>411</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QLabel" name="binSrchInfoConsoleLabel">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>310</y>
     <width>47</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Theory:</string>
   </property>
  </widget>
  <widget class="QPushButton" name="binSrchHomeBtn">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>10</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Home</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
